certified-mail-sender/src/main/scala/it/pagopa/interop/certifiedMailSender/JobExecution.scala

```diff
@@ -22,33 +22,49 @@ final case class JobExecution(sqsHandler: SQSHandler) {
     .ssl(true)()
 
   def run()(implicit blockingEC: ExecutionContextExecutor): Future[Unit] = sqsHandler.processAllRawMessages(
-    ApplicationConfiguration.maxNumberOfMessages,
+    ApplicationConfiguration.maxConsumeBatchSize,
     ApplicationConfiguration.visibilityTimeout
   ) { (messages, receipts) =>
-    logger.info(s"Sending ${messages.size} messages")
-    sendMessages(messages) >> deleteMessages(receipts)
+    logger.info(s"Processing ${messages.size} messages")
+    processMessage(messages) >>= deleteMessages(receipts)
   }
 
-  private def sendMessages(messages: List[String])(implicit ec: ExecutionContext): Future[Unit] =
-    Future.traverse(messages)(sendMail).void
+  private def processMessage(messages: List[String])(implicit ec: ExecutionContext): Future[List[InteropEnvelope]] =
+    Future.traverse(messages)(sendMail)
 
-  private def deleteMessages(receiptHandles: List[String])(implicit ec: ExecutionContext): Future[Unit] =
-    Future.traverse(receiptHandles)(sqsHandler.deleteMessage).void
+  private def deleteMessages(
+    receiptHandles: List[String]
+  )(implicit ec: ExecutionContext): List[InteropEnvelope] => Future[Unit] = envelopes => {
+    val deletions: List[(InteropEnvelope, String)] = envelopes.zip(receiptHandles)
+    Future
+      .traverse(deletions) { case (envelope, receipt) =>
+        logger.info(s"Deleting envelope ${envelope.id.toString} with receipt $receipt")
+        sqsHandler
+          .deleteMessage(receipt)
+          .recoverWith { ex =>
+            logger.error(
+              s"Error trying to delete envelope ${envelope.id.toString} with receipt $receipt - ${ex.getMessage}"
+            )
+            Future.failed(ex)
+          }
+      }
+      .void
```

Comment:
avoid using cats with futures :)

Reply:
@TonioGela ok, I would remove it from all jobs, are you agree?

Reply:
nope, non now, it will clash the PR I have prepared, for the moment I think it's enough to switch to `.map(_ => ())`